src/components/SignupForm.jsx

```diff
@@ -0,0 +1,165 @@
+/* eslint-disable react/require-default-props */
+/* eslint-disable jsx-a11y/label-has-for */
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React from 'react';
+import { connect } from 'react-redux';
+import propTypes from 'prop-types';
+import { Link, withRouter, Redirect } from 'react-router-dom';
+import { signUpUser, clearErrors, addError } from '../actions/signupActions';
+import ErrorAlert from '../views/ErrorAlert';
+
+/**
+ * @class Signup
+ * @extends { React }
+ */
+class Signup extends React.Component {
+  state = {
+    data: {
+      email: '',
+      password: '',
+      username: ''
+    },
+  }
+
+  /**
+   * This function handles the onsubmit event. It triggers validation and sends data to the API
+   * @param {object} e - This is the event object
+   */
+  handleSubmit = (e) => {
+    e.preventDefault();
+    const { signUpUser } = this.props;
+    const { data :{ email, password, username } } = this.state;
+    const userData = {
+      email: email,
+      password: password,
+      username: username,
+    };
+    if (this.handleValidation(userData)) {
+      signUpUser(userData);
+    }
+  }
+
+  /**
+   * This function validates data from the form
+   * @param {object} data - This is data gotten from the form
+   */
+  handleValidation = (data) => {
+    const { addError } = this.props;
+    let errors = [];
+    if(data.email == "") {
+      errors.push({message: "Email is required", field: "email"});
+      addError(errors);
+      return false;
+    }
+    if (data.username == "") {
+      errors.push({message: "A valid username is required", field: "username"});
+      addError(errors);
+      return false;
+    }
+    if ( data.password == "") {
+      errors.push({message: "A password is required", field: "password"});
+      addError(errors);
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * This function updates the state, as the data in the forms change
+   *  @param {object} e - This is the event object
+   */
+  handleChange = (e) => {
+    const { data } = { ...this.state };
+    data[e.currentTarget.name] = e.currentTarget.value;
+    this.setState({ data });
+  }; 
+
+  /**
+   * This function updates the errors in the store, it clears the current errors
+   */
+  handleErrorClose = (e) => {
+    const { clearErrors } = this.props;
+    clearErrors();
+  }
+
+  /**
+   * @returns {JSX} - returns the signup form
+   */
+  render(){
+    const { success, history } = this.props;
+    if (success) {
```

Comment:
#### Nitpick
- Aside from the `success` and `errors`, what other conditions are you looking out for, if none then refactor this to just one conditional block since it's just 2 instances you are looking at?